dev/annotate_source.py

```diff
@@ -170,27 +173,36 @@ def __partition_tags_on_pattern(self):
   def tag_head(self):
     """Tags the current branch's HEAD with the next semver tag.
     """
-    version_bump = self.determine_next_tag()
-    if version_bump is None:
-      # next_tag is None if we don't need to add a tag. There is already a
-      # 'version-X.Y.Z-$build' tag at HEAD.
-      logging.warn("There is already a tag of the form 'version-X.Y.Z-$build' at HEAD.")
-      return
-
-    next_tag = '{0}-{1}'.format(version_bump.version_str, self.__build_number)
-    # This tag is for logical identification for developers. This will be pushed
-    # to the upstream git repository if we choose to use this version in a
-    # formal Spinnaker product release.
-    run_quick('git -C {path} tag {next_tag} HEAD'
-              .format(path=self.__path, next_tag=next_tag))
+    if self.__is_head_current():
+      if self.__force_rebuild:
+        self.__tag_head_with_build(self.__current_version.tag)
+      else:
+        logging.warn("There is already a tag of the form 'version-X.Y.Z' at HEAD. Not forcing rebuild.")
+        return
+    else:
+      version_bump = self.determine_new_tag()
+      # This tag is for logical identification for developers. This will be pushed
+      # to the upstream git repository if we choose to use this version in a
+      # formal Spinnaker product release.
+      run_quick('git -C {path} tag {next_tag} HEAD'
+                .format(path=self.__path, next_tag=version_bump.version_str))
+      self.__tag_head_with_build(version_bump.version_str)
 
+  def __tag_head_with_build(self, version_bump_tag):
+    """Tags the current branch's HEAD with the next semver gradle build tag.
+
+    Args:
+      version_bump [VersionBump]: Semver to add as a gradle build tag.
+    """
+    next_tag_with_build = '{0}-{1}'.format(version_bump_tag,
+                                           self.__build_number)
     # This tag is for gradle to use as the package version. It incorporates the
     # build number for uniqueness when publishing. This tag is of the form
     # 'X.Y.Z-$build_number' for gradle to use correctly. This is not pushed
     # to the upstream git repository.
-    first_dash_idx = next_tag.index('-')
+    first_dash_idx = next_tag_with_build.index('-')
```

Comment:
Actually no it doesn't, that's a trivial check since there's always a string with at least '-'. I just removed that check.